src/update-feeds/update-feeds.ts

```diff
@@ -193,27 +195,61 @@ export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId) =>
       ...dapiResponse,
       shouldUpdate,
       signedData,
+      sponsorWallet,
     };
   });
 
-export const updateFeeds = async (_batch: ReturnType<typeof getFeedsToUpdate>) => {
+export const updateFeeds = async (chains: Chains, batch: ReturnType<typeof getFeedsToUpdate>) => {
   // TODO implement
   // batch, execute
+
+  // Prepare to send transactions with each provider on each chain
+  const promisedTransactions = batch.flatMap((item) => {
+    const { chainId, sponsorWallet } = item;
+    const { providers, gasSettings } = chains[chainId]!;
+
+    Object.entries(providers).map(async ([providerName, { url }]) => {
```

Comment:
This is not what we want to do. Each `runUpdateFeed` loop is run separately per provider. What we need to do is to pass the provider (or probably even better the contracts that are connected to the provider).